remote-wallet/src/remote_wallet.rs

```diff
@@ -322,20 +324,73 @@ impl RemoteWalletInfo {
     }
 }
 
+#[derive(Clone, Default, PartialEq)]
+pub struct DerivationPathComponent(u32);
+
+impl DerivationPathComponent {
+    pub const HARDENED_BIT: u32 = 1 << 31;
+
+    pub fn as_u32(&self) -> u32 {
+        self.0
+    }
+}
+
+impl From<u32> for DerivationPathComponent {
+    fn from(n: u32) -> Self {
+        Self(n | Self::HARDENED_BIT)
+    }
+}
+
+impl FromStr for DerivationPathComponent {
+    type Err = RemoteWalletError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Replace str::splitn() with str::strip_suffix() once stabilized
+        let parts: Vec<_> = s.splitn(2, '\'').collect();
+        if parts.len() == 2 {
+            eprintln!("all path components are promoted to hardened representation");
+        }
+        parts[0].parse::<u32>().map(|ki| ki.into()).map_err(|_| {
+            RemoteWalletError::InvalidDerivationPath(format!(
+                "failed to parse path component: {:?}",
+                s
+            ))
+        })
+    }
+}
+
+impl std::fmt::Display for DerivationPathComponent {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let hardened = if (self.0 & Self::HARDENED_BIT) == 0 {
+            ""
+        } else {
+            "'"
+        };
+        let index = self.0 & !Self::HARDENED_BIT;
+        write!(fmt, "{}{}", index, hardened)
+    }
+}
+
+impl std::fmt::Debug for DerivationPathComponent {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        std::fmt::Display::fmt(self, fmt)
+    }
+}
+
 #[derive(Default, PartialEq, Clone)]
 pub struct DerivationPath {
-    pub account: Option<u32>,
-    pub change: Option<u32>,
+    pub account: Option<DerivationPathComponent>,
+    pub change: Option<DerivationPathComponent>,
 }
 
 impl fmt::Debug for DerivationPath {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let account = if let Some(account) = self.account {
+        let account = if let Some(account) = &self.account {
             format!("/{:?}'", account)
```

Comment:
Can you please remove the apostrophe here and on 394?

Reply:
Done!  Added a test too.  https://github.com/solana-labs/solana/pull/10725/commits/2e6c29ff6c1c343b47244f533a86bb02140efe1d#diff-7ef79c2848cdef4bdf705e9d9000c5dfR648-R658